internal/source/git.go

```diff
@@ -126,8 +107,40 @@ func (r *Git) Unpack(ctx context.Context, bundle *rukpakv1alpha1.Bundle) (*Resul
 	return &Result{Bundle: bundleFS, ResolvedSource: resolvedSource, State: StateUnpacked}, nil
 }
 
+func (r *Git) createCloneOptions(bundle *rukpakv1alpha1.Bundle, progress bytes.Buffer) (*git.CloneOptions, error) {
```

Comment:
Should we refactor this method to be a function since there's no internal Git structure state being utilized in this function?